src/test/java/executor/service/service/ScenarioExecutorImplTest.java

```diff
@@ -0,0 +1,110 @@
+
+import static org.mockito.ArgumentMatchers.any;
+import static org.mockito.ArgumentMatchers.anyString;
+import static org.mockito.Mockito.*;
+import static org.mockito.Mockito.times;
+
+/**
+ * Class for testing the implementation of an ScenarioExecutorImpl.class.
+ *
+ * @author Oleksandr Tuleninov
+ * @version 01
+ */
+public class ScenarioExecutorImplTest {
+
+    private StepExecutionFabric stepExecutionFabric;
+    private StepExecution stepExecution;
+    private Scenario scenario;
+    private WebDriverConfig webDriverConfig;
+
+    @BeforeEach
+    public void setUp() {
+        this.stepExecutionFabric = mock(StepExecutionFabric.class);
+        this.stepExecution = mock(StepExecution.class);
+        this.scenario = getScenario();
+        this.webDriverConfig = getWebDriverConfig();
+    }
+
+    @AfterEach
+    void tearDown() {
+        this.stepExecutionFabric = null;
+        this.scenario = null;
+    }
+
+    @Test
+    public void testInvokeMethodExecute() {
+        ScenarioExecutor scenarioExecutor = mock(ScenarioExecutor.class);
+        doNothing().when(scenarioExecutor).execute(any(Scenario.class), any(WebDriver.class));
+
+        WebDriver webDriver = new WebDriverInitializerImpl(webDriverConfig).getInstance(getProxyConfigHolder());
+        scenarioExecutor.execute(scenario, webDriver);
+
+        verify(scenarioExecutor, times(1)).execute(any(Scenario.class), any(WebDriver.class));
+        verifyNoMoreInteractions(scenarioExecutor);
+    }
+
+    @Test
+    public void testWorkMethodExecute() {
+        when(stepExecutionFabric.getStepExecutor(anyString())).thenReturn(stepExecution);
+        doNothing().when(stepExecution).step(any(WebDriver.class), any(Step.class));
+
+        ScenarioExecutor executionService = new ScenarioExecutorImpl(stepExecutionFabric);
+        WebDriver webDriver = new WebDriverInitializerImpl(webDriverConfig).getInstance(getProxyConfigHolder());
+        executionService.execute(scenario, webDriver);
+
+        verify(stepExecution, times(numberOfSteps())).step(any(), any(Step.class));
+    }
+
+    private WebDriverConfig getWebDriverConfig() {
+        return new WebDriverConfig(
+                "chromedriver.exe",
+                "Mozilla/5.0 (Windows NT 10.0; Win64; x64) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/92.0.4515.131 Safari/537.36",
+                15000L,
+                5000L);
+    }
+
+    private List<Step> listOfSteps() {
+        return getScenario().getSteps();
+    }
+
+    private int numberOfSteps() {
+        return listOfSteps().size();
+    }
+
+    private Scenario getScenario() {
+        return new Scenario(
+                "test scenario 1",
+                "http://info.cern.ch",
+                Arrays.asList(
+                        new Step(StepTypes.CLICK_CSS, "body > ul > li:nth-child(1) > a"),
+                        new Step(StepTypes.SLEEP, "5000:15000"),
+                        new Step(StepTypes.CLICK_XPATH, "/html/body/p")));
```

Comment:
can be an empty scenario, since you are using Mockito.any()

Reply:
I disagree because I do **verify(stepExecution, times(numberOfSteps())).step(any(), any(Step.class));**